src/Playwright.Tests/BrowserContextStorageStateTests.cs

```diff
@@ -109,5 +109,38 @@ public async Task ShouldRoundTripThroughTheFile()
             Assert.AreEqual("value1", await page2.EvaluateAsync<string>("localStorage['name1']"));
             Assert.AreEqual("username=John Doe", await page2.EvaluateAsync<string>("document.cookie"));
         }
+
+        [PlaywrightTest("browsercontext-storage-state.spec.ts", "should capture cookies")]
+        public async Task ShouldCaptureCookies()
+        {
+            Server.SetRoute("/setcookie.html", context =>
+            {
+                context.Response.Cookies.Append("a", "b");
+                context.Response.Cookies.Append("empty", "");
+                return Task.CompletedTask;
+            });
+
+            await Page.GotoAsync(Server.Prefix + "/setcookie.html");
+            CollectionAssert.AreEqual(new[] { "a=b", "empty=" }, await Page.EvaluateAsync<string[]>(@"() =>
+            {
+                const cookies = document.cookie.split(';');
+                return cookies.map(cookie => cookie.trim()).sort();
+            }"));
+
+            var storageState = await Context.StorageStateAsync();
+            StringAssert.Contains(@"""name"":""a"",""value"":""b""", storageState);
+            StringAssert.Contains(@"""name"":""empty"",""value"":""""", storageState);
+            StringAssert.Contains(@"""sameSite"":""Lax""", storageState);
+            StringAssert.DoesNotContain(@"""url"":null", storageState);
```

Comment:
Added these two asserts to check for the behavior change which are not included upstream: https://github.com/microsoft/playwright/blob/71a4f69acf3e827402ba208939c677ea4a38bca4/tests/browsercontext-storage-state.spec.ts#L107-L143